diff -up libical-0.43/src/libical/autogenex/ical.h.fix-ical-errors-are-fatal-checks libical-0.43/src/libical/autogenex/ical.h
--- libical-0.43/src/libical/autogenex/ical.h.fix-ical-errors-are-fatal-checks	2009-01-08 11:50:21.000000000 -0500
+++ libical-0.43/src/libical/autogenex/ical.h	2012-10-08 16:44:23.655810465 -0400
@@ -3272,7 +3272,7 @@ if(icalerror_get_error_state(x)==ICAL_ER
 void icalerror_set_errno(icalerrorenum x);
 #endif
 
-#ifdef ICAL_ERRORS_ARE_FATAL
+#if ICAL_ERRORS_ARE_FATAL == 1
 #undef NDEBUG
 #endif
 
@@ -3282,7 +3282,7 @@ void icalerror_set_errno(icalerrorenum x
 #define icalerror_check_component_type(value,type);
 
 /* Assert with a message */
-#ifdef ICAL_ERRORS_ARE_FATAL
+#if ICAL_ERRORS_ARE_FATAL == 1
 
 #ifdef __GNUC__
 #define icalerror_assert(test,message) if(!(test)){fprintf(stderr,"%s(), %s:%d: %s\n",__FUNCTION__,__FILE__,__LINE__,message);icalerror_stop_here(); abort();}
diff -up libical-0.43/src/libical/icalerror.h.fix-ical-errors-are-fatal-checks libical-0.43/src/libical/icalerror.h
--- libical-0.43/src/libical/icalerror.h.fix-ical-errors-are-fatal-checks	2009-01-08 11:50:22.000000000 -0500
+++ libical-0.43/src/libical/icalerror.h	2012-10-08 16:44:23.655810465 -0400
@@ -111,7 +111,7 @@ if(icalerror_get_error_state(x)==ICAL_ER
 void icalerror_set_errno(icalerrorenum x); 
 #endif
 
-#ifdef ICAL_ERRORS_ARE_FATAL
+#if ICAL_ERRORS_ARE_FATAL == 1
 #undef NDEBUG
 #endif
 
@@ -121,7 +121,7 @@ void icalerror_set_errno(icalerrorenum x
 #define icalerror_check_component_type(value,type);
 
 /* Assert with a message */
-#ifdef ICAL_ERRORS_ARE_FATAL
+#if ICAL_ERRORS_ARE_FATAL == 1
 
 #ifdef __GNUC__
 #define icalerror_assert(test,message) if(!(test)){fprintf(stderr,"%s(), %s:%d: %s\n",__FUNCTION__,__FILE__,__LINE__,message);icalerror_stop_here(); abort();}
diff -up libical-0.43/src/libical/icalparser.c.fix-ical-errors-are-fatal-checks libical-0.43/src/libical/icalparser.c
--- libical-0.43/src/libical/icalparser.c.fix-ical-errors-are-fatal-checks	2009-01-08 11:50:21.000000000 -0500
+++ libical-0.43/src/libical/icalparser.c	2012-10-08 16:44:23.657810868 -0400
@@ -917,10 +917,7 @@ icalcomponent* icalparser_add_line(icalp
 
                 /* Change for mozilla */
                 /* have the option of being flexible towards unsupported parameters */
-                #ifndef ICAL_ERRORS_ARE_FATAL
-                continue;
-                #endif
-
+		#if ICAL_ERRORS_ARE_FATAL == 1
 		insert_error(tail, str, "Cant parse parameter name",
 			     ICAL_XLICERRORTYPE_PARAMETERNAMEPARSEERROR);
 		tail = 0;
@@ -934,6 +931,9 @@ icalcomponent* icalparser_add_line(icalp
 			name = 0;
 		}
 		return 0;
+		#else
+		continue;
+		#endif
 	    }
 
 	    /* if (pvalue) {
